File: Java/app/src/main/java/kz/sgq/fs_tiktaktoe/mvp/model/interfaces/FieldModel.java
package kz.sgq.fs_tiktaktoe.mvp.model.interfaces;

import java.util.List;

public interface FieldModel {
    String getScorePlayerOne();
    String getScorePlayerTwo();
    String getStep();
    boolean isMode();
    boolean onClickBox(int i);
    boolean isStep();
    boolean checkVictory();
    boolean checkDraw();
    boolean getEnd();
    int stepBot();
    List<Integer> getBoxWinList();
    void setStep(boolean b);
    void setMode(boolean b);
    void setStepText(String str);
    void createLists();
    void clearScore();
}

File: Kotlin/app/src/main/java/kz/sgq/kotlintictactoe/mvp/model/interfaces/FieldModel.kt
package kz.sgq.kotlintictactoe.mvp.model.interfaces

interface FieldModel {
    fun getScorePlayerOne(): String
    fun getScorePlayerTwo(): String
    fun getStep(): String
    fun onClickBox(i: Int): Boolean
    fun isStep(): Boolean
    fun isMode(): Boolean
    fun checkVictory(): Boolean
    fun checkDraw(): Boolean
    fun getEnd(): Boolean
    fun stepBot(): Int
    fun getBoxWinList(): MutableList<Int>
    fun setStep(boolean: Boolean)
    fun setMode(mode: Boolean)
    fun createLists()
    fun clearScore()
    fun setStepText(string: String)
}